### ISSUE TITLE
Include views in a different context

### ISSUE THREAD

**Issue by jvorcak** [bennylope/django-organizations#144]
Hi,

First of all thank you for your hard work and this plugin. It's awesome.
However, I'm missing some documentation about whether/how it's possible to include default view functionality you provide in my templates.

Let's say I like the default behaviour of your views -> `/accounts` && `/accounts/1/` and I'd like to include it to my page (to a specific part of a template). Is there any easy way to do so?

NOTE: I've came back to Django after 1-2 years, so the solution might be obvious and maybe it's out of scope of this plugin's documentation, but maybe it could help if it's more explicit.

**Comment by bennylope** [comment]
Hi Ján - I'm not sure what you mean by "include [the default behavior of the views] in [your] page". Do you mean using your own template with the views here?

It _sounds like_ you're asking how to either (1) use a different template or (2) add data into the view.

For the first, there are two ways to do this: either (a) pass a `template_name` variable to the view factory method (`as_view`) in your URL conf (like [this in the docs](https://docs.djangoproject.com/en/2.0/topics/class-based-views/#simple-usage-in-your-urlconf)) or subclass the particular view in your own `views.py` module, using that class, and overriding the `template_name` definition in your class (like [this in the docs](https://docs.djangoproject.com/en/2.0/topics/class-based-views/#subclassing-generic-views)). All things being equal the second method is preferred.

If you mean how do you include other data or functionality, then continue with the last recommendation and create a subclass of the view. You can add whatever data or methods you want here.

```python
class MyAccountView(OrganizationList):
    template_name = "myapp/account_list.html"

    def get_context_data(self, **kwargs):
        context = super().get_context_data(**kwargs)
        context.update({
            'is_this_real_life': self.is_this_real_life(),  # defined elsewhere on the class
            ...
        })
        return context
```

Does that answer your question?

**Comment by jvorcak** [comment]
Yes thank you, it partially does. 

But I was looking for more like an out of a box solution where I could just override base template and use a plugin without a need to override any View.

 ```
 url(r'^accounts/', include('organizations.urls')),
 url(r'^invitations/', include(invitation_backend().get_urls())),
```

**Comment by bennylope** [comment]
> But I was looking for more like an out of a box solution where I could just override base template and use a plugin without a need to override any View.

So something like the dependency injection used by the invitation/registration backends? While technically feasible, that's neither present in django-organizations nor is that really how Django CBVs work or at least are expected to work. I think you'd be best off overriding the view. It'll be a little bit of code and it'll be explicit to boot.

**Comment by jvorcak** [comment]
Thanks a lot for your help :)

**Comment by jvorcak** [comment]
Hi,

So at the end I've achieved my goal by doing this:

1) I've created `organizations_base.html` file
```
{% extends 'base.html' %}
{% block content %}
{% endblock content %}
```

And I've overriden templates in a `organizations` folder based on my needs. https://github.com/bennylope/django-organizations/tree/dev/organizations/templates

That way I don't need to create anything special (no views, etc), I've just overriden the base template + templates I need. I hope this will help somebody.

Thanks for your time & help @bennylope !
